components/thread/ThreadPost.tsx

```diff
@@ -148,143 +247,92 @@ const ThreadPost: React.FC<ThreadPostProps> = ({
+    [post.postId, avatarRef]
   );
 
-  const callbackRef = (postRef: PostHandler | undefined | null) => {
-    if (postRef) {
-      // TODO: also remove this ref
-      addPostHandlerRef({ postId: post.postId, ref: postRef });
-    }
-    if (!extraProps.avatarRef) {
-      return;
-    }
-    if (typeof extraProps.avatarRef === "function") {
-      extraProps.avatarRef(postRef?.avatarRef?.current || null);
-    } else {
-      // @ts-ignore
-      extraProps.avatarRef.current = postRef?.avatarRef?.current || null;
-    }
-  };
-  const { forceHideIdentity } = useForceHideIdentity();
+  const postsData = useGetPostsData({
+    post,
+    showPostAncestors,
+    showThreadStarter,
+  });
 
-  const memoizedPropsMap = React.useMemo(() => {
-    return posts.map((post): PostProps => {
-      const directLink = cachedLinks.getLinkToPost({
-        slug,
-        threadId,
-        postId: post.postId,
-      });
-      return {
-        createdTime: formatDistanceToNow(new Date(post.created), {
-          addSuffix: true,
-        }),
-        text: post.content,
-        secretIdentity: post.secretIdentity,
-        userIdentity: post.userIdentity,
-        createdTimeLink: {
-          href: `${directLink.href}${getCurrentSearchParams()}`,
-          onClick: directLink.onClick,
-        },
-        notesLink: threadLink,
-        onNewContribution: onNewContributionCallback,
-        onNewComment: onNewCommentCallback,
-        totalComments: postCommentsMap.get(post.postId)?.total || 0,
-        directContributions: parentChildrenMap.get(post.postId)?.children
-          .length,
-        totalContributions: getTotalContributions(post, parentChildrenMap),
-        newPost: post.isNew,
-        newComments: postCommentsMap.get(post.postId)?.new || 0,
-        newContributions: isLoggedIn
-          ? getTotalNewContributions(post, parentChildrenMap)
-          : 0,
-        tags: post.tags,
-        allowsComment: realmPermissions.includes(
-          RealmPermissions.COMMENT_ON_REALM
-        ),
-        allowsContribution: realmPermissions.includes(
-          RealmPermissions.POST_ON_REALM
-        ),
-        menuOptions: options,
-        op: post.secretIdentity.name == opIdentity?.name,
-        getOptionsForTag: tagOptions,
-        ...extraProps,
-      };
-    });
-  }, [
-    posts,
-    cachedLinks,
-    extraProps,
-    onNotesClick,
-    isLoggedIn,
-    realmPermissions,
-    // onNewCommentCallback,
-    // onNewContributionCallback,
-    // onNotesClickWithId,
-    // options,
-    // parentChildrenMap,
-    // slug,
-    // threadId,
-    // threadLink,
-  ]);
+  // This link in the notes may have a different action than the link in the
+  // date display if a special notes click action has been passed from above
+  const notesLink = React.useMemo(
+    () => ({
+      href: `${
+        postsData[postsData.length - 1].createdTimeLink.href
+      }${getCurrentSearchParams()}`,
+      onClick: onNotesClick
```

Comment:
Having the href present even when the notes button collapses the sub-thread instead of being a link is weird, and a bit confusing, because when you hover the button, the href still shows at the bottom of the browser as if clicking it will take you to the post page.

Reply:
Addressed this and the other concern in the review comment together. I chose better variable names (thanks for the suggestion), and removed the href if we have an onclick. A future solution would be to let consumers pass down the full `LinkWithAction`, but this is good enough for now.